Lib/argparse.py

```diff
@@ -873,9 +873,6 @@ def __init__(self,
                 option_string = '--no-' + option_string[2:]
                 _option_strings.append(option_string)
 
-        if help is not None and default is not None:
```

Comment:
I think the effect of this edit is too broad.  Instead, focus on making sure the default only prints once when the _formatter_class_ is set to *ArgumentDefaultsHelpFormatter*.  When _formatter_class_ isn't set, the default still needs to be shown.

Given this script:

```
from argparse import *

parser = ArgumentParser()
parser.add_argument("--foo", action=BooleanOptionalAction, help="Whether to foo it", default=False)
parser.add_argument("--quux", help="Set the quux", default=42)
print(parser.parse_args())
```

We still want this output:
```
$ py tmp.py --help
usage: tmp.py [-h] [--foo | --no-foo] [--quux QUUX]

optional arguments:
  -h, --help       show this help message and exit
  --foo, --no-foo  Whether to foo it (default: False)
  --quux QUUX      Set the quux
```

Reply:
Hi I have updated the code in add action function since it is not possible to get formatter class in action object, ArgumentDefaultsHelpFormatter already add default value and that was the origin problem which result in duplicate default value in the help string. Now with the default formatter class it can also show the default value.

Reply:
Hey @rhettinger could you speak on why the default should be displayed for BooleanOptionalArgument out-of-the-box whereas all other shipped actions require setting `formatter_class=ArgumentDefaultsHelpFormatter`?  I do find that showing the default value is useful but it seems to break the expected separation of responsibility between action and formatter.

I ask about this since I have a different bug which would be fixed if BooleanOptionalArgument were to no longer try to add defaults on its own.  If BooleanOptionalArgument should continue to do that formatting, though, I could extract the code from  ArgumentDefaultsHelpFormatter and have both ArgumentDefaultsHelpFormatter and BooleanOptionalArgument use it which would likely fix my issue, this issue, and a similar class of issues that we just haven't encountered yet.